feat: add C ndarray interface and refactor implementation for `stats/base/snanmeanpn`

0PrashantYadav0 opened the pull request:
Resolves None.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C ndarray interface for `stats/base/snanmeanpn`
-   refactors JavaScript implementation
-   updates tests, docs, examples and benchmarks

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

0PrashantYadav0 committed:
feat: updated stats/base/snanmeanpn native addon from C++ to C

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@aayush0325 Will you provide a code review for this pr?

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/src/main.c:
```diff
@@ -36,11 +37,25 @@
+}
+
+/**
+* Computes the arithmetic mean of a single-precision floating-point strided array, ignoring `NaN` values and using a two-pass error correction algorithm.
```
```suggestion
/**
* Computes the arithmetic mean of a single-precision floating-point strided array, ignoring `NaN` values and using a two-pass error correction algorithm and alternative indexing semantics.
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
````diff
@@ -180,6 +161,107 @@ console.log( v );
+Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array, ignoring `NaN` values and using a two-pass error correction algorithm and alternative indexing semantics.
+
+```c
+const double x[] = { 1.0, -2.0, NaN, 2.0 };
````
```suggestion
const double x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
````diff
@@ -180,6 +161,107 @@ console.log( v );
+Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array `x`, ignoring `NaN` values and using a two-pass error correction algorithm.
+
+```c
+const double x[] = { 1.0, -2.0, NaN, 2.0 };
````
```suggestion
const double x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
```diff
@@ -154,22 +145,12 @@ var v = snanmeanpn.ndarray( N, x, 2, 1 );
-    if ( randu() < 0.2 ) {
-        x[ i ] = NaN;
-    } else {
-        x[ i ] = round( (randu()*100.0) - 50.0 );
-    }
-}
+var x = discreteUniform( 10, -50, 50, {
+    'dtype': 'float64'
+});
 console.log( x );
 
 var v = snanmeanpn( x.length, x, 1 );
```
please refer to #4781 to see how we plan to do JavaScript examples for `NaN` related packages

aayush0325 replied:
same for `examples/index.js`

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/benchmark/benchmark.js:
kindly see how we're doing benchmarks for `NaN` related packages in [stats/base/dnanmax/benchmark](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/dnanmax/benchmark). Similarly for all other JavaScript and C benchmarks.

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/benchmark/benchmark.native.js:
same comment regarding benchmarks

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/benchmark/benchmark.ndarray.js:
same comment regarding benchmarks

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/benchmark/benchmark.ndarray.native.js:
same comment regarding benchmarks

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/benchmark/c/benchmark.length.c:
same comment regarding benchmarks

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/docs/repl.txt:
```diff
@@ -22,8 +22,8 @@
-    stride: integer
-        Index increment.
+    strideX: integer
+        Stride Length.
```
```suggestion
        Stride length.
```

aayush0325 replied:
i had been making this mistake for a while too, my bad!! we should take care of the grammar here and not capitalize the 2nd word

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/docs/repl.txt:
```diff
@@ -69,10 +66,10 @@
-    stride: integer
-        Index increment.
+    strideX: integer
+        Stride Length.
```
```suggestion
        Stride length.
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/examples/index.js:
kindly see #4781 for context

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/src/main.c:
```diff
@@ -1,7 +1,7 @@
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2025 The Stdlib Authors.
```
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```
Git isn't counting this as a new file like it does in some of the other PRs, no need to update the copyright year here

aayush0325 reviewed (changes requested):
Thanks for the PR, this needs some changes to match the work that we've been doing with `NaN` related packages in the stats namespace. Kindly see [stats/base/dnanmax](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/dnanmax/) as a refrence package.

0PrashantYadav0 committed:
chore: updated according to code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: failed
---

0PrashantYadav0 committed:
chore: updated float with double

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: failed
---

0PrashantYadav0 committed:
chore: updated benchmark files

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: failed
---

0PrashantYadav0 committed:
chore: updated ndarray input type

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@aayush0325 i tried to make the changes you asked for. Please let me know if i have made a mistake anywhere.

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
````diff
@@ -180,6 +168,107 @@ console.log( v );
+```c
+const double x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
+
+double v = stdlib_strided_dnanmeanpn_ndarray( 3, x, 1, 0 );
````
```suggestion
double v = stdlib_strided_dnanmeanpn_ndarray( 4, x, 1, 0 );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
````diff
@@ -180,6 +168,107 @@ console.log( v );
+```c
+const double x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
+
+double v = stdlib_strided_dnanmeanpn( 3, x, 1 );
````
```suggestion
double v = stdlib_strided_dnanmeanpn( 4, x, 1 );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/benchmark/benchmark.js:
```diff
@@ -21,16 +21,30 @@
+* @returns {number} random number or `NaN`
+*/
+function rand() {
+	if ( bernoulli( 0.2 ) ) {
```
```suggestion
	if ( bernoulli( 0.8 ) < 1 ) {
```
can you please do this instead? this is the convention that we're going to follow from now on.

aayush0325 replied:
same comment for all js benchmarks

0PrashantYadav0 committed:
chore: updated input value

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
Ok updated the changes you asked for. @aayush0325

aayush0325 reviewed (approved):
lgtm

0PrashantYadav0 commented:
@kgryte @Planeshifter Please review and merge this pr

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
````diff
@@ -180,6 +168,107 @@ console.log( v );
+### Usage
+
+```c
+#include "stdlib/stats/base/dnanmeanpn.h"
````
```suggestion
#include "stdlib/stats/base/snanmeanpn.h"
```

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
````diff
@@ -180,6 +168,107 @@ console.log( v );
+#include "stdlib/stats/base/dnanmeanpn.h"
+```
+
+#### stdlib_strided_dnanmeanpn( N, \*X, strideX )
````
```suggestion
#### stdlib_strided_snanmeanpn( N, \*X, strideX )
```

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
````diff
@@ -180,6 +168,107 @@ console.log( v );
+Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array `x`, ignoring `NaN` values and using a two-pass error correction algorithm.
+
+```c
+const double x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
````
These should be floats. In short, you need to fix the copy-paste mistakes throughout the C documentation here.

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
```diff
@@ -180,6 +168,107 @@ console.log( v );
+
+int main( void ) {
+    // Create a strided array:
+    const float x[] = { 1.0, 2.0, 0.0/0.0, 3.0, 0.0/0.0, 4.0, 5.0, 6.0, 0.0/0.0, 7.0, 8.0, 0.0/0.0 };
```
You're missing `f` suffixes.

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/benchmark/c/benchmark.length.c:
```diff
@@ -94,19 +94,24 @@ static float rand_float( void ) {
 
 	for ( i = 0; i < len; i++ ) {
-		x[ i ] = ( rand_float()*20000.0f ) - 10000.0f;
+		if ( rand_float() < 0.2 ) {
+			x[ i ] = 0.0 / 0.0; // NaN
+		} else {
+			x[ i ] = ( rand_float() * 20000.0 ) - 10000.0;
```
```suggestion
		if ( rand_float() < 0.2f ) {
			x[ i ] = 0.0f / 0.0f; // NaN
		} else {
			x[ i ] = ( rand_float() * 20000.0f ) - 10000.0f;
```

This applies here and below.

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/docs/repl.txt:
```diff
@@ -89,10 +86,8 @@
     ~-0.3333
 
     See Also
     --------
```
```suggestion
    --------

```

No reason to delete this empty line. While not wholly consistent, we typically include in our repl help files to ensure proper spacing in the REPL.

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/examples/c/example.c:
```diff
@@ -17,21 +17,20 @@
 int main( void ) {
 	// Create a strided array:
-	float x[] = { 1.0, 2.0, 0.0/0.0, 3.0, 0.0/0.0, 4.0, 5.0, 6.0, 0.0/0.0, 7.0, 8.0, 0.0/0.0 };
+	const float x[] = { 1.0, 2.0, 0.0/0.0, 3.0, 0.0/0.0, 4.0, 5.0, 6.0, 0.0/0.0, 7.0, 8.0, 0.0/0.0 };
```
You're missing `f` suffixes.

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/examples/index.js:
```diff
@@ -18,22 +18,19 @@
+var uniform = require( '@stdlib/random/base/uniform' );
+var filledarrayBy = require( '@stdlib/array/filled-by' );
+var bernoulli = require( '@stdlib/random/base/bernoulli' );
+var snanmeanpn = require( '@stdlib/stats/base/snanmeanpn' );
```
```suggestion
var snanmeanpn = require( './../lib' );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/lib/index.js:
```diff
@@ -28,20 +28,17 @@
-* var floor = require( '@stdlib/math/base/special/floor' );
 * var snanmeanpn = require( '@stdlib/stats/base/snanmeanpn' );
 *
 * var x = new Float32Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN ] );
```
```suggestion
* var x = new Float32Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN, NaN ] );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/lib/index.js:
```diff
@@ -28,20 +28,17 @@
-* var N = floor( x.length / 2 );
 *
-* var v = snanmeanpn.ndarray( N, x, 2, 1 );
+* var v = snanmeanpn.ndarray( 4, x, 2, 1 );
```
```suggestion
* var v = snanmeanpn.ndarray( 5, x, 2, 1 );
```

The point of the example should be to include a `NaN` value among the iterated elements.

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/lib/ndarray.js:
```diff
@@ -39,21 +39,19 @@ var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
-* var N = floor( x.length / 2 );
 *
-* var v = snanmeanpn( N, x, 2, 1 );
+* var v = snanmeanpn( 4, x, 2, 1 );
```
Same comment.

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/lib/ndarray.native.js:
```diff
@@ -31,27 +30,20 @@ var addon = require( './snanmeanpn.native.js' );
-* var N = floor( x.length / 2 );
 *
-* var v = snanmeanpn( N, x, 2, 1 );
+* var v = snanmeanpn( 4, x, 2, 1 );
```
Same comment.

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/src/addon.c:
```diff
@@ -34,10 +35,27 @@
-	STDLIB_NAPI_CREATE_DOUBLE( env, (double)stdlib_strided_snanmeanpn( N, X, stride ), v );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 1 );
+	STDLIB_NAPI_CREATE_DOUBLE( env, API_SUFFIX(stdlib_strided_snanmeanpn)( N, X, strideX ), v );
```
```suggestion
	STDLIB_NAPI_CREATE_DOUBLE( env, (double)API_SUFFIX(stdlib_strided_snanmeanpn)( N, X, strideX ), v );
```

The return value is a `float`, so we need/want an explicit cast.

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/src/addon.c:
```diff
@@ -34,10 +35,27 @@
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_INT64( env, offsetX, argv, 3 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 1 );
+	STDLIB_NAPI_CREATE_DOUBLE( env, API_SUFFIX(stdlib_strided_snanmeanpn_ndarray)( N, X, strideX, offsetX ), v );
```
```suggestion
	STDLIB_NAPI_CREATE_DOUBLE( env, (double)API_SUFFIX(stdlib_strided_snanmeanpn_ndarray)( N, X, strideX, offsetX ), v );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/src/main.c:
```diff
@@ -49,14 +64,10 @@ float stdlib_strided_snanmeanpn( const int64_t N, const float *X, const int64_t
 	}
-	if ( N == 1 || stride == 0 ) {
+	if ( N == 1 || strideX == 0 ) {
 		return X[ 0 ];
```
```suggestion
		return X[ offsetX ];
```

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/test/test.ndarray.js:
```diff
@@ -106,15 +104,13 @@ tape( 'the function supports a `stride` parameter', function test( t ) {
 
-	N = floor( x.length / 2 );
-	v = snanmeanpn( N, x, 2, 0 );
+	v = snanmeanpn( 4, x, 2, 0 );
```
```suggestion
	v = snanmeanpn( 5, x, 2, 0 );
```

The comments are the source of truth. You should revisit the tests to ensure that we are providing the correct number of values based on what the comments indicate. Here, that is 5 values.

kgryte reviewed (changes requested):
Left an initial round of comments.

cc @aayush0325

0PrashantYadav0 committed:
chore: updated input value and added float value

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 committed:
chore: updated text inputs

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: failed
---

0PrashantYadav0 committed:
chore: updated text inputs

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

0PrashantYadav0 commented:
@kgryte i don't change the value of input `N` in the test `the function supports a negative `stride` parameter'` in `test.ndarray.js` and `test.ndarray.native.js`. If i change value of `N` i have to change offset value or input value to match the expected value. Also `x.length/2` will provide value as 4. I hope i resolved everything please review and merge this pr. Thank you.

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/test/test.ndarray.js:
```diff
@@ -130,8 +126,7 @@ tape( 'the function supports a negative `stride` parameter', function test( t )
 
-	N = floor( x.length / 2 );
-	v = snanmeanpn( N, x, -2, 6 );
+	v = snanmeanpn( 4, x, -2, 6 );
```
```suggestion
	v = snanmeanpn( 5, x, -2, 8 );
```
see the commented indexes which show how the array is being traversed, same for `test.ndarray.native.js`

0PrashantYadav0 committed:
fix: updated test for negative stride

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

0PrashantYadav0 commented:
@aayush0325 Thank you for follow up.

aayush0325 commented:
can you please make changes here as done in #4994

0PrashantYadav0 commented:
Done 👍

0PrashantYadav0 committed:
fix: update examples to use correct N values and handle NaN in arrays

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/src/main.c:
```diff
@@ -36,11 +37,25 @@
 * @param stride  stride length
```
```suggestion
* @param strideX  stride length
```

aayush0325 reviewed (changes requested):
minor comment, otherwise lgtm

0PrashantYadav0 committed:
fix: rename stride parameter for clarity in snanmeanpn function

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
```diff
@@ -180,6 +172,107 @@ console.log( v );
+<!-- C usage documentation. -->
+
+<section class="usage">
+
```
the structure of the README here is off, see [an example README](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/stats/strided/dmax/README.md?plain=1#L149) to see how it should look like. please check for these in all your other open PRs as well I'd appreciate it very much!

0PrashantYadav0 commented:
@aayush0325 Made the changes you asked for please review.

0PrashantYadav0 committed:
fix: improve documentation for snanmeanpn function

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 reviewed (dismissed):
lgtm

0PrashantYadav0 committed:
chore: minor changes

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

0PrashantYadav0 commented:
@kgryte Good to go. please review and merge.

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
```diff
@@ -51,84 +51,79 @@ The [arithmetic mean][arithmetic-mean] is defined as
 -   **N**: number of indexed elements.
 -   **x**: input [`Float32Array`][@stdlib/array/float32].
--   **stride**: index increment for `x`.
+-   **strideX**: index increment for `x`.
```
```suggestion
-   **strideX**: stride length for `x`.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
```diff
@@ -180,6 +172,127 @@ console.log( v );
+
+#### stdlib_strided_snanmeanpn( N, \*X, strideX )
+
+Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array `x`, ignoring `NaN` values and using a two-pass error correction algorithm.
```
```suggestion
Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array, ignoring `NaN` values and using a two-pass error correction algorithm.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
```diff
@@ -51,84 +51,79 @@ The [arithmetic mean][arithmetic-mean] is defined as
-#### snanmeanpn( N, x, stride )
+#### snanmeanpn( N, x, strideX )
 
 Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array `x`, ignoring `NaN` values and using a two-pass error correction algorithm.
```
```suggestion
Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array, ignoring `NaN` values and using a two-pass error correction algorithm.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanpn/README.md:
```diff
@@ -180,6 +172,127 @@ console.log( v );
+
+<!-- /.c -->
+
+<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->
+
+<section class="related">
+
```
```suggestion
```

kgryte committed:
docs: fix markup

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
